web/static/models/endpoint-model.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Enum for Globus endpoint entity types
+ * @readonly
+ * @enum {string}
+ */
+const EndpointEntityType = Object.freeze({
+  GCP_MAPPED_COLLECTION: 'GCP_mapped_collection',
+  GCP_GUEST_COLLECTION: 'GCP_guest_collection',
+  GCSV5_ENDPOINT: 'GCSv5_endpoint',
+  GCSV5_MAPPED_COLLECTION: 'GCSv5_mapped_collection',
+  GCSV5_GUEST_COLLECTION: 'GCSv5_guest_collection',
+  GCSV4_HOST: 'GCSv4_host',
+  GCSV4_SHARE: 'GCSv4_share',
+
+  /**
+   * Check if entity type is a mapped collection
+   * @param {string} type - Entity type to check
+   * @returns {boolean} True if the entity type is a mapped collection
+   */
+  isMappedCollection(type) {
+    return type === this.GCP_MAPPED_COLLECTION ||
+      type === this.GCSV5_MAPPED_COLLECTION;
+  },
+
+  /**
+   * Check if entity type is a guest collection
+   * @param {string} type - Entity type to check
+   * @returns {boolean} True if the entity type is a guest collection
+   */
+  isGuestCollection(type) {
+    return type === this.GCP_GUEST_COLLECTION ||
+      type === this.GCSV5_GUEST_COLLECTION;
+  },
+
+  /**
+   * Check if entity requires consent
+   * According to docs, only GCSv5 mapped collections require consent
+   * @param {string} type - Entity type to check
+   * @returns {boolean} True if the entity type requires consent
+   */
+  requiresConsent(type) {
+    return type === this.GCSV5_MAPPED_COLLECTION;
+  }
+});
+
+/**
+ * Model class for Globus endpoint data and operations
+ */
+class EndpointModel {
+  #endpoint;
+  #displayName;
+  #entityType;
+  #collections;
+  #ownerString;
+  #isConnected;
+  #isBusy;
+
+  /**
+   * @param {object} endpointData - Endpoint data from Globus API
+   */
+  constructor(endpointData = {}) {
+    this.#endpoint = endpointData;
+    this.#displayName = endpointData.display_name || endpointData.canonical_name || "Unknown Endpoint";
+    this.#entityType = endpointData.entity_type || null;
+    this.#collections = endpointData.collections || [];
+    this.#ownerString = this.#formatOwnerString(endpointData);
+    this.#isConnected = endpointData.is_connected === true;
+    this.#isBusy = endpointData.in_use === true;
+  }
```

Comment:
My preference here would be to raise an error, if the endpointData is invalid or unknown. i.e. if you are missing keys.